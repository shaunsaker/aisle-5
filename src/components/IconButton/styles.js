import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const SIZE = 50;
const SMALL_SIZE = 24;

const styles = StyleSheet.create({
  container: {
    width: SIZE,
    height: SIZE,
    borderRadius: SIZE,
    backgroundColor: styleConstants.colors.secondary,
    justifyContent: 'center',
    alignItems: 'center',
    ...styleConstants.shadows.large,
  },
  smallContainer: {
    width: SMALL_SIZE,
    height: SMALL_SIZE,
    borderRadius: SMALL_SIZE,
  },
  secondaryContainer: {
    backgroundColor: styleConstants.colors.accent,
  },
  icon: {
    fontSize: 30,
    color: styleConstants.colors.white,
  },
  smallIcon: {
    fontSize: 16,
  },
});

export default styles;
