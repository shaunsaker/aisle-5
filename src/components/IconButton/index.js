import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  name: PropTypes.string,
  handlePress: PropTypes.func,
  small: PropTypes.bool,
  secondary: PropTypes.bool,
};

const defaultProps = {};

const IconButton = ({ name, handlePress, small, secondary }) => {
  return (
    <Touchable
      onPress={handlePress}
      style={[
        styles.container,
        small && styles.smallContainer,
        secondary && styles.secondaryContainer,
      ]}
    >
      <Icon
        name={name}
        style={[styles.icon, small && styles.smallIcon, secondary && styles.secondaryIcon]}
      />
    </Touchable>
  );
};

IconButton.propTypes = propTypes;
IconButton.defaultProps = defaultProps;

export default IconButton;
