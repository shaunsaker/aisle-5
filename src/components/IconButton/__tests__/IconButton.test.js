import React from 'react';
import renderer from 'react-test-renderer';

import IconButton from '..';

describe('IconButton', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<IconButton />);

      expect(component).toMatchSnapshot();
    });

    it('renders the small state', () => {
      const component = renderer.create(<IconButton small />);

      expect(component).toMatchSnapshot();
    });

    it('renders the secondary state', () => {
      const component = renderer.create(<IconButton secondary />);

      expect(component).toMatchSnapshot();
    });
  });
});
