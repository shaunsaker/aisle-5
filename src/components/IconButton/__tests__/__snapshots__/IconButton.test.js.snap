// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders renders the secondary state 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": null,
      "borderRadius": 50,
      "borderWidth": 0,
      "elevation": 12,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 4,
      "width": 50,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#EDE8CA",
            "fontSize": 30,
          },
          undefined,
          Object {
            "color": "#08293A",
          },
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`IconButton renders renders the small state 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FEA116",
      "borderColor": null,
      "borderRadius": 30,
      "borderWidth": 0,
      "elevation": 6,
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 2,
      "width": 30,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#EDE8CA",
            "fontSize": 30,
          },
          Object {
            "fontSize": 20,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`IconButton renders renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FEA116",
      "borderColor": null,
      "borderRadius": 50,
      "borderWidth": 0,
      "elevation": 12,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 4,
      "width": 50,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#EDE8CA",
            "fontSize": 30,
          },
          undefined,
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
